### ISSUE TITLE
Calling the Healthcheck kill the app

### ISSUE THREAD

**Issue by pierresouchay** [jippi/hashi-ui#364]
OS: Mac OS High Sierra
Version 0.22.0 hashi-ui-darwin-amd64  (md5: 27b33d5713f18c8d1db52b9ee9c05a0d)

Launched with:
```
hashi-ui-darwin-amd64 --consul-enable --consul-address my.remote.consul.server:8500 --listen-address 0.0.0.0:3001 --consul-color \#FF8F1C --consul-read-only true
```

While the GUI works (but take a huge amount of CPUs on large clusters), the healthcheck kill the app: Just after launching the app, simply call:
```
curl -i localhost:3001/_status
curl: (52) Empty reply from server
```

On the server side, the message is the following:
```
[...]
INFO[0000]
INFO[0000] Connecting to Consul ...
INFO[0000] done!
INFO[0000] Listening ...
2017/11/17 00:38:01 http: panic serving [::1]:62840: runtime error: invalid memory address or nil pointer dereference
goroutine 25 [running]:
net/http.(*conn).serve.func1(0xc42000ff40)
	/home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:1697 +0xd0
panic(0x16067a0, 0x1d9b6e0)
	/home/travis/.gimme/versions/go1.9.linux.amd64/src/runtime/panic.go:491 +0x283
main.StatusHandler.func1(0x1a16180, 0xc42042e0e0, 0xc42016a700)
	/home/travis/gopath/src/github.com/jippi/hashi-ui/backend/handler_status.go:63 +0x473
net/http.HandlerFunc.ServeHTTP(0xc4204cc260, 0x1a16180, 0xc42042e0e0, 0xc42016a700)
	/home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:1918 +0x44
github.com/jippi/hashi-ui/backend/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc420166b90, 0x1a16180, 0xc42042e0e0, 0xc42016a700)
	/home/travis/gopath/src/github.com/jippi/hashi-ui/backend/vendor/github.com/gorilla/mux/mux.go:114 +0xdc
net/http.serverHandler.ServeHTTP(0xc4200821a0, 0x1a16180, 0xc42042e0e0, 0xc42016a500)
	/home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2619 +0xb4
net/http.(*conn).serve(0xc42000ff40, 0x1a16ec0, 0xc420154140)
	/home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:1801 +0x71d
created by net/http.(*Server).Serve
	/home/travis/.gimme/versions/go1.9.linux.amd64/src/net/http/server.go:2720 +0x288
```

using `--log-level trace` do not give more information

**Comment by jippi** [comment]
Thanks for the report :)

**Comment by jippi** [comment]
Oh, what is a "large" cluster and what is a "huge amount of resources"? :)

**Comment by pierresouchay** [comment]
@jippi Actually, I made a test yesterday with remote and degraded network conditions (and I did not had a clear look).

On the resources side, it is not that bad (I thought I had around 100MB/sec of network trafic, but I was wrong, it is more in the 3-4MB/sec)

The cluster is round 5k nodes per DC (and connected to 7 similar clusters), with around 1300 different kind of services.

I quickly tried it, but actually, was was slowing down really was the browser (maybe refresh rate is a bit aggressive by default)

As soon as the healthcheck will work, I'll have a serious try.

Thank you
